src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/PDFSI.cs

```diff
@@ -151,14 +151,25 @@ private async Task<DataElement> StorePDF(Stream pdfStream, Instance instance)
             if (instance.Title?["nb"] != null && instance.Title?["nb"] != String.Empty)
             {
                 fileName = instance.Title["nb"] + ".pdf";
+                fileName = GetValidFileName(fileName);
             }
 
             return await _dataService.InsertBinaryData(
                 instance.Id,
                 pdfElementType,
                 "application/pdf",
                 fileName ?? defaultFileName,
-                pdfStream);            
+                pdfStream);
+        }
+
+        private string GetValidFileName(string fileName)
+        {
+            foreach (char c in System.IO.Path.GetInvalidFileNameChars())
```

Comment:
If possible, don't use namespace here, put the namespace up with the rest of the using statements.